lib/util/source-code.js

```diff
@@ -114,6 +115,12 @@ function SourceCode(text, ast) {
      */
     this.lines = text.split(/\r\n|\r|\n|\u2028|\u2029/g);
 
+    /**
+     * Fixes to be made later on.
+     * @type Object[]
+     */
+    this._fixes = [];
```

Comment:
On one hand I see how queuing up fixes and applying them in batches might prevent collisions, on the other hand, if one rule fixes an issue, which in turn will cause another rule to fail, failure will not be reported... Not sure if it's good or bad. This might cause a situation where you run eslint with fix flag and get an all clear, and then run it again, without any changes and get warnings/errors.